src/components/MessageList/utils/filterTypingUsers.ts

```diff
@@ -0,0 +1,64 @@
+import type { ChatContextValue } from '../../../contexts/chatContext/ChatContext';
+import type { TypingContextValue } from '../../../contexts/typingContext/TypingContext';
+import type { ThreadContextValue } from '../../../contexts/threadContext/ThreadContext';
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../../types/types';
+
+type FilterTypingUsersParams<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = Pick<TypingContextValue<At, Ch, Co, Ev, Me, Re, Us>, 'typing'> &
+  Pick<ChatContextValue<At, Ch, Co, Ev, Me, Re, Us>, 'client'> &
+  Pick<ThreadContextValue<At, Ch, Co, Ev, Me, Re, Us>, 'thread'>;
+
+export const filterTypingUsers = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+>({
+  client,
+  thread,
+  typing,
+}: FilterTypingUsersParams<At, Ch, Co, Ev, Me, Re, Us>) => {
+  const typingKeys = Object.keys(typing);
+  const nonSelfUsers: string[] = [];
+  typingKeys.forEach((typingKey) => {
```

Comment:
Maybe use `for ... in` instead? You won't need Object.keys separately

Reply:
I think that would iterate over the inherited keys and we have a lint rule that doesnt allow to check own properties

![image](https://user-images.githubusercontent.com/18744505/115907366-27a5bc00-a43f-11eb-9ff5-3ce3941a9fb4.png)

Reply:
> A for...in loop only iterates over enumerable, non-Symbol properties. Objects created from built–in constructors like Array and Object have inherited non–enumerable properties from Object.prototype and String.prototype, such as String's indexOf() method or Object's toString() method.

So I guess you won't need hasOwnProperties!! Correct me if I am wrong plz